test/integration/auth/auth.prose.test.ts

```diff
@@ -0,0 +1,416 @@
+            try {
+              const stats = await client.db('test').stats();
+              expect(stats).to.exist;
+            } finally {
+              await client.close();
+            }
+          }
+        );
+
+        it(
+          `authenticates ${user.description} using mechanism negotiaton and url`,
+          metadata,
+          async function () {
+            const username = encodeURIComponent(user.username);
+            const password = encodeURIComponent(user.password);
+            const url = makeConnectionString(this.configuration, username, password);
+
+            const client = this.configuration.newClient(url);
+            try {
+              const stats = await client.db('test').stats();
+              expect(stats).to.exist;
+            } finally {
+              await client.close();
+            }
+          }
+        );
+      }
+
+      /**
+       * Step 2
+       * For a test user supporting both SCRAM-SHA-1 and SCRAM-SHA-256, drivers should verify
+       * that negotation selects SCRAM-SHA-256. This may require monkey patching, manual log
+       * analysis, etc.
+       * todo(NODE-5629): Test passes locally but will fail on CI runs.
+       */
+      it.skip(
+        'selects SCRAM-SHA-256 for a user that supports both auth mechanisms',
+        metadata,
+        async function () {
+          const options = {
+            auth: {
+              username: userMap.both.username,
+              password: userMap.both.password
+            },
+            authSource: this.configuration.db
+          };
+
+          const spy = sinon.spy(ScramSHA256.prototype, 'auth');
+          const client = this.configuration.newClient({}, options);
+          try {
+            const stats = await client.db('test').stats();
+            expect(stats).to.exist;
+            expect(spy.called).to.equal(true);
+          } finally {
+            sinon.restore();
+            await client.close();
+          }
+        }
+      ).skipReason = 'todo(NODE-5629): Test passes locally but will fail on CI runs.';
+
+      /**
+       * Step 3
+       * For test users that support only one mechanism, verify that explictly specifying
+       * the other mechanism fails.
+       */
+      it(
+        'fails to connect if incorrect auth mechanism (SCRAM-SHA-1) is explicitly specified',
+        metadata,
+        async function () {
+          const options = {
+            auth: {
+              username: userMap.sha256.username,
+              password: userMap.sha256.password
+            },
+            authSource: 'admin',
+            authMechanism: 'SCRAM-SHA-1'
+          };
+
+          const client = this.configuration.newClient({}, options);
+          const error = await client
+            .db('test')
+            .stats()
+            .catch(e => e);
+          expect(error.message).to.match(/Authentication failed|SCRAM/);
+          await client.close();
```

Comment:
We use `finally` in a bunch of other tests to make sure we close the client, why not here? 

In my opinion, it would be easier to ensure this cleanup if we assign the client to a suite level `let` and use afterEach. If that sounds preferable you have to put the tests that are in the for loop into a function so they can remain bound to the same client they create

Reply:
I've put the close calls now into `afterEach`, and moved the client to the top level to avoid needing to extract the tests in the for loop into a function.